upload/upload.go

```diff
@@ -126,23 +130,34 @@ func uploadFiles(files, outFiles []string, targetDir string, config *helpers.Con
 		}
 		_ = f.Close()
 	}
+	s3Endpoint := config.HostBase
+	if !strings.HasPrefix(s3Endpoint, "http") {
+		s3Endpoint = fmt.Sprintf("http://%s", config.HostBase)
+		if !config.UseHTTPS {
+			s3Endpoint = fmt.Sprintf("https://%s", config.HostBase)
+		}
+	}
```

Comment:
This has been done at https://github.com/NBISweden/sda-cli/blob/main/helpers/helpers.go#L240-L242. If you want to improve the logic, modify there.